sherpa/astro/ui/tests/test_astro_supports_pha2.py

```diff
@@ -39,6 +39,16 @@
 from sherpa.astro.data import DataPHA
 
 
+@pytest.fixture(autouse=True)
```

Comment:
I have an alternate version in https://github.com/sherpa/sherpa/pull/412/commits/35c228774669ec932b92da35f4cd73ebe8487fb2 which doesn't require request. I don't understand if one is better than the other, or whether there's any difference in behavior or what versions they work in or ...

Reply:
From the `pytest` documentation:

https://docs.pytest.org/en/latest/fixture.html#fixture-finalization-executing-teardown-code
> Note that if an exception happens during the setup code (before the yield keyword), the teardown code (after the yield) will not be called.
> An alternative option for executing teardown code is to make use of the addfinalizer method of the request-context object to register finalization functions.

That's why I prefer to be explicit with the finalizer, although in this case the likelihood of `ui.clean()` throwing an exception is rather low, and anyway in that case not executing `ui.clean()` during tear down doesn't seem to make any difference.

Reply:
On the other hand, I think this fixture should probably be autoused for all tests, and while I have the temptation to make it a global fixture, we should probably make it `autouse=True` at least for new tests. That also prevents test functions from having a dangling, unused parameter.

Reply:
One issue with `autouse=True` is that there are plenty of tests which don't use the session interface, so it's just extra work. Then again, it shouldn't be a significant run-time cost. I would also suggest in this case that it calls both `sherpa.ui.clean()` and `sherpa.astro.ui.clean()` - since they refer to different session objects. I don't think this needs to be done or decided upon to accept this PR.